internal/pkg/admission/policy-server-secret.go

```diff
@@ -42,31 +42,22 @@ func (r *Reconciler) fetchOrInitializePolicyServerSecret(ctx context.Context) (*
 	return &policyServerSecret, nil
 }
 
-func (r *Reconciler) buildPolicyServerSecret() (*corev1.Secret, error) {
-	ca, caPrivateKey, err := admissionregistration.GenerateCA()
+var generateCA = admissionregistration.GenerateCA
+var pemEncodeCertificate = admissionregistration.PemEncodeCertificate
```

Comment:
I didn't get it in the beginning, but then I saw the unit test :heart_eyes: :clap:

Reply:
It's nice. I would recommend passing a function as an argument to the functions that rely on this variable. This way there's no implicit hidden behavior that depends on a global package variable being set to different values.

Reply:
Yes, I agree on that. Another possibility is to make these variables part of the `Reconciler` struct.

One final note: please change their names to have the `Func` suffix, it makes it easier to understand what they are about

Reply:
ok, I created package vars because in my opinion makes the code easier to read. But I get your point that someone else inside the package could potentially change the behavior. I changed it in the last commit, please have a look.
Thanks for the feedback!